addons/event/models/event_registration.py

```diff
@@ -270,6 +271,20 @@ def _update_mail_schedulers(self):
         # cron triggers are correctly used
         onsubscribe_schedulers.execute()
 
+    def action_unarchive(self):
+        """ Set active=True on a recordset, by calling toggle_active to
+        take the corresponding actions according to the model, after checking
+        seat availability"""
+        n_seats_to_add = Counter(self.mapped("event_id.id"))
+
+        if any(registration.event_id.seats_limited
+               and registration.event_id.seats_max
+               and registration.event_id.seats_available < n_seats_to_add[registration.event_id.id]
+               for registration in self):
+            raise ValidationError(_('No more available seats for this event.'))
```

Comment:
I think a "UserError" is more appropriate here.
ValidationErrors are usually used when validating field values.

Reply:
I used that one for consistency. Should I update the error raised when trying to save a registration for fully booked event too, or it's OK if they are different?

Reply:
Oh I see, the other one is raised when trying to save the form in a "@api.constrains" block though so ValidationError makes sense in that context.

But here I think UserError is a bit better and suits the context more.